src/components/search/SearchTypesDirectives.js

```diff
@@ -0,0 +1,395 @@
+(function() {
+  goog.provide('ga_search_type_directives');
+
+  goog.require('ga_browsersniffer_service');
+  goog.require('ga_debounce_service');
+  goog.require('ga_layer_metadata_popup_service');
+  goog.require('ga_map_service');
+  goog.require('ga_marker_overlay_service');
+  goog.require('ga_search_service');
+  goog.require('ga_urlutils_service');
+
+  var originToZoomLevel = {
+    address: 10,
+    parcel: 10,
+    sn25: 8
+  };
+
+  var parseExtent = function(stringBox2D) {
+    var extent = stringBox2D.replace('BOX(', '')
+      .replace(')', '').replace(',', ' ')
+      .split(' ');
+    return $.map(extent, parseFloat);
+  };
+
+  var addOverlay = function(gaOverlay, map, res) {
+    var visible = originToZoomLevel.hasOwnProperty(res.attrs.origin);
+    var center = [res.attrs.y, res.attrs.x];
+    if (!res.attrs.y || !res.attrs.x) {
+      center = ol.proj.transform([res.attrs.lon, res.attrs.lat],
+          'EPSG:4326', 'EPSG:21781');
+    }
+    gaOverlay.add(map,
+                  center,
+                  parseExtent(res.attrs.geom_st_box2d),
+                  visible);
+
+  };
+
+  var removeOverlay = function(gaOverlay, map) {
+    gaOverlay.remove(map);
+  };
+
+  var listenerMoveEnd;
+  var registerMove = function(gaOverlay, gaDebounce, map) {
+    listenerMoveEnd = map.on('moveend', gaDebounce.debounce(function() {
+      var zoom = map.getView().getZoom();
+      gaOverlay.setVisibility(zoom);
+    }, 200, false, false));
+  };
+
+  var unregisterMove = function() {
+    if (listenerMoveEnd) {
+      listenerMoveEnd.src.unByKey(listenerMoveEnd);
```

Comment:
The ol3 api suggest we should use static function instead http://openlayers.org/en/master/apidoc/ol.Observable.html#unByKey